File: anvil/tests/fixtures/cpp/good_code.cpp
/**
 * Good C++ code sample.
 *
 * This file demonstrates well-formatted, standards-compliant C++ code
 * that should pass all validators.
 */

#include <iostream>
#include <string>
#include <vector>

namespace example {

/**
 * Calculate the sum of numbers in a vector.
 *
 * @param numbers Vector of integers to sum
 * @return Sum of all numbers
 */
int calculate_sum(const std::vector<int>& numbers) {
    int sum = 0;
    for (int num : numbers) {
        sum += num;
    }
    return sum;
}

/**
 * DataProcessor class for processing data with configurable options.
 */
class DataProcessor {
 public:
    /**
     * Constructor.
     *
     * @param name Name of the processor
     * @param debug Enable debug output
     */
    DataProcessor(const std::string& name, bool debug = false)
        : name_(name), debug_(debug) {}

    /**
     * Add an item to the data list.
     *
     * @param item Item to add
     */
    void add_data(const std::string& item) {
        data_.push_back(item);
        if (debug_) {
            std::cout << "Added item: " << item << std::endl;
        }
    }

    /**
     * Get the number of items.
     *
     * @return Number of items in the data list
     */
    size_t get_count() const {
        return data_.size();
    }

 private:
    std::string name_;
    bool debug_;
    std::vector<std::string> data_;
};

}  // namespace example

int main() {
    example::DataProcessor processor("example", true);
    processor.add_data("item1");
    processor.add_data("item2");
    std::cout << "Total items: " << processor.get_count() << std::endl;
    return 0;
}

File: anvil/tests/fixtures/python/good_code.py
"""
Good Python code sample.

This module demonstrates well-formatted, PEP 8 compliant Python code
that should pass all validators.
"""

import sys
from pathlib import Path
from typing import List, Optional


def calculate_sum(numbers: List[int]) -> int:
    """
    Calculate the sum of a list of numbers.

    Args:
        numbers: List of integers to sum

    Returns:
        The sum of all numbers in the list
    """
    return sum(numbers)


def process_file(file_path: str) -> Optional[str]:
    """
    Read and process a file.

    Args:
        file_path: Path to the file to process

    Returns:
        File contents if successful, None otherwise
    """
    path = Path(file_path)
    if not path.exists():
        return None

    try:
        with open(path, "r", encoding="utf-8") as f:
            content = f.read()
        return content.strip()
    except IOError as e:
        print(f"Error reading file: {e}", file=sys.stderr)
        return None


class DataProcessor:
    """Process data with configurable options."""

    def __init__(self, name: str, debug: bool = False):
        """
        Initialize the processor.

        Args:
            name: Name of the processor
            debug: Enable debug output
        """
        self.name = name
        self.debug = debug
        self._data: List[str] = []

    def add_data(self, item: str) -> None:
        """Add an item to the data list."""
        self._data.append(item)
        if self.debug:
            print(f"Added item: {item}")

    def get_count(self) -> int:
        """Get the number of items."""
        return len(self._data)


if __name__ == "__main__":
    # Example usage
    processor = DataProcessor("example", debug=True)
    processor.add_data("item1")
    processor.add_data("item2")
    print(f"Total items: {processor.get_count()}")
